platform/darwin/src/MGLStyle.mm

```diff
@@ -308,7 +308,7 @@ - (void)removeObjectFromLayersAtIndex:(NSUInteger)index
         [NSException raise:NSRangeException
```

Comment:
If you pass in the `index` of the last layer in the array, wouldn’t `index` be equal to `layers.size() - 1`, triggering this exception?